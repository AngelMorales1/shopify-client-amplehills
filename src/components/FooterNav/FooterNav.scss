@import 'settings', 'src/styles/typography';

.FooterNav {
  &__link-container {
    flex-wrap: wrap;

    @media ($medium-up) {
      flex-wrap: nowrap;
    }
  }

  &__link {
    width: 50%;

    @media ($medium-up) {
      width: auto;
    }
  }
}
