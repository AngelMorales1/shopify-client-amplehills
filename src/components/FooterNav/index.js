import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import RoutesWithoutFooterExtras from 'constants/RoutesWithoutFooterExtras';
import get from 'utils/get';

class FooterNav extends PureComponent {
  routeOmitsFooterNav = () => {
    return RoutesWithoutFooterExtras.includes(this.props.pathname);
  };

  render() {
    if (this.routeOmitsFooterNav()) return null;

    const items = Object.values(get(this.props, 'items.simpleFragments', {}));
    const defaultIcon = '/assets/images/bubble-icon.svg';
    if (!items.length) return null;

    return (
      <div className="xs-hide sm-hide col-12 drip bg-white z-1 relative">
        <div className="container-width mx-auto flex py3">
          {items.map(item => (
            <div
              key={get(item, 'uuid')}
              className="w100 flex flex-column center"
            >
              <a
                className="text-decoration-none hover-slide-up"
                href={get(item, 'link')}
              >
                <img
                  className="px4 w100"
                  src={get(item, 'icon.data', defaultIcon)}
                  alt={get(item, 'icon.name', '')}
                />
                <p className="py3 text-peach bold">{get(item, 'text')}</p>
              </a>
            </div>
          ))}
        </div>
      </div>
    );
  }
}

FooterNav.propTypes = {
  pathname: PropTypes.string,
  items: PropTypes.oneOfType(
    PropTypes.arrayOf(
      PropTypes.shape({
        simpleFragments: PropTypes.objectOf(
          PropTypes.shape({
            icon: PropTypes.shape({
              data: PropTypes.string,
              name: PropTypes.name
            }),
            uuid: PropTypes.string,
            link: PropTypes.string,
            text: PropTypes.string
          })
        )
      })
    ),
    PropTypes.object
  )
};

FooterNav.defaultProps = {
  pathname: '',
  items: [
    {
      simpleFragments: {}
    }
  ]
};

export default FooterNav;
