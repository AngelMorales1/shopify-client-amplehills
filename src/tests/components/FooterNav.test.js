import React from 'react';
import { shallow } from 'enzyme';

import FooterNav from 'components/FooterNav';

it('renders without data', () => {
  const component = shallow(<FooterNav />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const items = [
    {
      icon: {
        data: '',
        name: 'foo'
      },
      link: '/foo',
      text: 'Foo',
      uuid: '12345'
    }
  ];
  const component = shallow(<FooterNav pathname={'/foo'} items={items} />);

  expect(component).toMatchSnapshot();
});
