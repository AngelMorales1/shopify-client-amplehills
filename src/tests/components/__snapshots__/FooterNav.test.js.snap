// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FooterNav
    items={
        Array [
            Object {
              "icon": Object {
                "data": "",
                "name": "foo",
              },
              "link": "/foo",
              "text": "Foo",
              "uuid": "12345",
            },
          ]
    }
    pathname="/foo"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FooterNav
    items={
        Array [
            Object {
              "simpleFragments": Object {},
            },
          ]
    }
    pathname=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
